distribution/helm/vector-aggregator/templates/_helpers.tpl

```diff
@@ -1,18 +1,39 @@
 {{/* vim: set filetype=mustache: */}}
 
 {{/*
-Resolve effective service ports to use.
+Internal template to render service ports depending on whether service is a headless service or not. Use
+either 'vector-aggregator.servicePorts' or 'vector-aggregator.headlessServicePorts' as entry points.
 */}}
-{{- define "vector-aggregator.servicePorts" -}}
-{{- if .Values.vectorSource.enabled }}
-- name: vector
-  port: {{ .Values.vectorSource.listenPort }}
-  protocol: TCP
-  targetPort: {{ .Values.vectorSource.listenPort }}
+{{- define "vector-aggregator.internalServicePorts" -}}
+{{- $headless := index . 0 -}}
+{{- $values := index . 1 -}}
+{{- if $values.vectorSource.enabled }}
+{{- $servicePort := merge (dict) $values.vectorSource -}}
+{{- $_ := set $servicePort "name" "vector" -}}
+{{- $_ := set $servicePort "port" $values.vectorSource.listenPort -}}
+{{- $_ := set $servicePort "protocol" "TCP" -}}
+{{- if not $headless -}}
+{{- $_ := set $servicePort "targetPort" $values.vectorSource.listenPort -}}
+{{- end -}}
```

Comment:
```suggestion
{{- $servicePort := dict -}}
{{- $_ := set $servicePort "name" "vector" -}}
{{- $_ := set $servicePort "port" $values.vectorSource.listenPort -}}
{{- $_ := set $servicePort "protocol" "TCP" -}}
{{- $_ := set $servicePort "targetPort" $values.vectorSource.listenPort -}}
```

I'm pretty sure we don't want to inherit the `$values.vectorSource` here, but to construct the service port explicitly. Also, no need to conditionally hide the target port for headless service, as it's handled at the `include "libvector.servicePort"` (which was the point for this change).

Reply:
Yeah, sure - I actually did discuss with myself there also :-) Just need to include the `nodePort` also as that is the only values that is inherited. I'll change that one.